configure.ac

```
@@ -2129,6 +2129,29 @@ if test "$Py_BOLT" = 'true' ; then
   else
     AC_MSG_ERROR([merge-fdata is required for a --enable-bolt build but could not be found.])
   fi
+
+  py_bolt_icf_flag="-icf=safe"
+  AC_CACHE_CHECK(
+    [whether ${LLVM_BOLT} supports safe identical code folding],
+    [py_cv_bolt_icf_safe],
+    [
+      saved_cflags="$CFLAGS"
+      saved_ldflags="$LDFLAGS"
+      CFLAGS="$CFLAGS_NODIST"
+      LDFLAGS="$LDFLAGS_NODIST"
+      AC_LINK_IFELSE(
+        [AC_LANG_PROGRAM([[]], [[]])],
+        [py_cv_bolt_icf_safe=no
+         ${LLVM_BOLT} -icf=safe -o conftest.bolt conftest$EXEEXT >&AS_MESSAGE_LOG_FD 2>&1 dnl
+         && py_cv_bolt_icf_safe=yes],
+        [AC_MSG_FAILURE([could not compile empty test program])])
+      CFLAGS="$saved_cflags"
+      LDFLAGS="$saved_ldflags"
```

Consider using our `WITH_SAVE_ENV` macro; it automatically saves and restores `CFLAGS`, `CPPFLAGS`, `LDFLAGS`, and `LIBS`:

```suggestion
      WITH_SAVE_ENV([
        CFLAGS="$CFLAGS_NODIST"
        LDFLAGS="$LDFLAGS_NODIST"
        AC_LINK_IFELSE(
          [AC_LANG_PROGRAM([], [])],
          [py_cv_bolt_icf_safe=no
           ${LLVM_BOLT} -icf=safe -o conftest.bolt conftest$EXEEXT >&AS_MESSAGE_LOG_FD 2>&1 dnl
           && py_cv_bolt_icf_safe=yes],
          [AC_MSG_FAILURE([could not compile empty test program])])
      ])
```

Moreover, the double brackets (`[[]]`) are not needed; a single pair (`[]`) is sufficient.